File: src/components/Recipe.js
import React from "react";

function Recipe(props) {
    const recipe = props.recipe;
    const ingredients = recipe.ingredients.map(ingredient =>
        <li key={ingredient}>{ingredient}</li>
    );

    const directions = (
        <div className={'content'}>
            <ol>
                {recipe.directions.map((direction) => <li>{direction}</li>)}
            </ol>
        </div>
    );

    return (

        <>
            <div id={'ingredients-column'} className={'column is-one-quarter'}>
                <div key={recipe.name}>
                    <h3 className='subtitle'>Ingredients:</h3>

                <div className={'content'}>
                    <ul>
                        {ingredients}
                    </ul>
                </div>

                </div>
            </div>

            <div id={'directions-column'} className={'column'}>
                <div key={recipe.name}>
                    <h3 className={'subtitle'}>Directions:</h3>
                    <p>{directions}</p>
                </div>
            </div>
        </>
    );
}

export default Recipe;

File: src/main/java/net/ehicks/recipe/beans/Recipe.java
package net.ehicks.recipe.beans;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Entity
@Table
public class Recipe implements Serializable
{
    @Id
    @GeneratedValue
    private Long id;

    private String name;
    private String emoji;
    private String description;
    private int difficulty;
    private String cookingTime;
    private int servings;

    @ManyToMany
    private List<Ingredient> ingredients;

    @ManyToMany
    private List<Direction> directions;

    public Recipe()
    {
    }

    @Override
    public boolean equals(Object obj)
    {
        if (!(obj instanceof User)) return false;
        User that = (User) obj;
        return this.id.equals(that.getId());
    }

    @Override
    public int hashCode()
    {
        return 17 * 37 * id.intValue();
    }

    public String toString()
    {
        return this.getClass().getSimpleName() + ":" + id;
    }

    // -------- Getters / Setters ----------

    public Long getId()
    {
        return id;
    }

    public void setId(Long id)
    {
        this.id = id;
    }

    public String getName()
    {
        return name;
    }

    public void setName(String title)
    {
        this.name = title;
    }

    public String getEmoji()
    {
        return emoji;
    }

    public void setEmoji(String emoji)
    {
        this.emoji = emoji;
    }

    public String getDescription()
    {
        return description;
    }

    public void setDescription(String description)
    {
        this.description = description;
    }

    public int getDifficulty()
    {
        return difficulty;
    }

    public void setDifficulty(int difficulty)
    {
        this.difficulty = difficulty;
    }

    public String getCookingTime()
    {
        return cookingTime;
    }

    public void setCookingTime(String cookingTime)
    {
        this.cookingTime = cookingTime;
    }

    public List<Ingredient> getIngredients()
    {
        return ingredients;
    }

    public void setIngredients(List<Ingredient> ingredients)
    {
        this.ingredients = ingredients;
    }

    public List<Direction> getDirections()
    {
        return directions;
    }

    public void setDirections(List<Direction> directions)
    {
        this.directions = directions;
    }

    public int getServings()
    {
        return servings;
    }

    public void setServings(int servings)
    {
        this.servings = servings;
    }
}
